websites/A/AniWatch/presence.ts

```diff
@@ -0,0 +1,264 @@
+				);
+			}
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "Join Room",
+						url: href,
+					},
+				];
+			}
+		}
+	} else if (pathname.startsWith("/watch")) {
+		const title = document.querySelector<HTMLDataListElement>(
+				"li.breadcrumb-item.dynamic-name.active"
+			),
+			episode = document.querySelector<HTMLSpanElement>(
+				"a.ep-item.active div.ssli-order"
+			),
+			thumbnail = document.querySelector<HTMLImageElement>(
+				".anis-watch-detail > .anis-content > .anisc-poster > .film-poster > img.film-poster-img"
+			)?.src;
+
+		presenceData.largeImageKey = thumbnail;
+		if (title) presenceData.details = title.textContent;
+		if (episode) presenceData.state = `Episode ${episode.textContent}`;
+		if (data && !data.paused) {
+			[, presenceData.endTimestamp] = presence.getTimestamps(
+				data.currTime,
+				data.duration
+			);
+			presenceData.smallImageKey = Assets.Play;
+		} else if (data) presenceData.smallImageKey = Assets.Pause;
+
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "Watch Episode",
+					url: href,
+				},
+			];
+		}
+	} else if (pathname.startsWith("/event/")) {
+		const title = document.querySelector<HTMLDivElement>("div.title"),
+			description = document.querySelector<HTMLDivElement>("div.description");
+		if (title) presenceData.details = `Event: ${title.textContent}`;
+		if (description) presenceData.state = description.textContent;
+	} else if (pathname.startsWith("/community")) {
+		presenceData.details = "Community Post";
+		presenceData.state = document.title;
+		presenceData.smallImageKey = Assets.Question;
+	} else {
+		switch (pathname) {
+			case "/search": {
+				presenceData.details = "Searching";
+
+				const query = document.querySelector(
+					"#main-content > section > div.block_area-header.block_area-header-tabs > div.float-left.bah-heading.mr-4 > h2 > i"
+				)?.textContent;
+
+				presenceData.state = query;
+				presenceData.smallImageKey = Assets.Search;
+				break;
+			}
+			case "/events": {
+				presenceData.details = "Looking at events";
+				break;
+			}
+			case "/contact": {
+				presenceData.details = "Contact Us";
+				break;
+			}
+			case "/dmca": {
+				presenceData.details = "DMCA";
+				break;
+			}
+			case "/terms": {
+				presenceData.details = "Terms of Service";
+				break;
+			}
+			case "/user/profile": {
+				const username = document
+					.querySelector(
+						"#main-wrapper > div.profile-header > div.container > div.ph-title"
+					)
+					.textContent.substring(3);
+				presenceData.details = `Viewing profile ${username}`;
+				presenceData.smallImageKey = Assets.Viewing;
+
+				break;
+			}
+			case "/user/settings": {
+				presenceData.details = "Changing Settings";
+				presenceData.smallImageKey = "https://i.imgur.com/AuOsdzi.png";
+
+				break;
+			}
+			case "/user/notification": {
+				presenceData.details = "Looking at Notifications";
+				presenceData.smallImageKey = "https://i.imgur.com/TjzBYj9.png";
```

Comment:
Considering the many assets used, add these links to the `Assets` enum:
https://github.com/PreMiD/Presences/blob/1f5a4117f92c6549a75e1c01c6305f9c4398b5b5/websites/A/Aodesu/presence.ts#L6-L14